extensions/azurecore/src/account-provider/azureAccountProvider2.ts

```diff
@@ -0,0 +1,443 @@
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as http from 'http';
+import * as url from 'url';
+import * as crypto from 'crypto';
+import * as nls from 'vscode-nls';
+import * as request from 'request';
+import {
+	AzureAccount,
+	AzureAccountProviderMetadata,
+	AzureAccountSecurityTokenCollection,
+	AzureAccountSecurityToken,
+	Tenant,
+} from './interfaces';
+
+import TokenCache from './tokenCache';
+import { AddressInfo } from 'net';
+import { AuthenticationContext, TokenResponse, ErrorResponse } from 'adal-node';
+import { promisify } from 'util';
+import * as events from 'events';
+
+const localize = nls.loadMessageBundle();
+const notInitalizedMessage = localize('accountProviderNotInitialized', "Account provider not initialized, cannot perform action");
+
+export class AzureAccountProvider implements azdata.AccountProvider {
+	private static AzureAccountAuthenticatedEvent: string = 'AzureAccountAuthenticated';
+	private static WorkSchoolAccountType: string = 'work_school';
+	private static MicrosoftAccountType: string = 'microsoft';
+	private static AadCommonTenant: string = 'common';
+
+	private static eventEmitter = new events.EventEmitter();
+	private static redirectUrlAAD = 'https://vscode-redirect.azurewebsites.net/';
+	private commonAuthorityUrl: string;
+	private isInitialized: boolean = false;
+
+
+	constructor(private metadata: AzureAccountProviderMetadata, private _tokenCache: TokenCache) {
+		this.commonAuthorityUrl = url.resolve(this.metadata.settings.host, AzureAccountProvider.AadCommonTenant);
+	}
+
+	// interface method
+	initialize(storedAccounts: azdata.Account[]): Thenable<azdata.Account[]> {
+		return this._initialize(storedAccounts);
+	}
+
+	private async _initialize(storedAccounts: azdata.Account[]): Promise<azdata.Account[]> {
+		for (let account of storedAccounts) {
+			try {
+				const x = await this.getAccessTokens(account, azdata.AzureResource.ResourceManagement);
+				console.log(x);
+			} catch (e) {
+				console.error(`Refreshing account ${account.displayInfo} failed - ${e}`);
+				account.isStale = true;
+				azdata.accounts.accountUpdated(account);
+			}
+		}
+		this.isInitialized = true;
+		return storedAccounts;
+	}
+
+	private async getToken(userId: string, tenantId: string, resourceId: string): Promise<TokenResponse> {
+		let authorityUrl = url.resolve(this.metadata.settings.host, tenantId);
+		const context = new AuthenticationContext(authorityUrl, null, this._tokenCache);
+
+		const acquireToken = promisify(context.acquireToken).bind(context);
+
+		let response: (TokenResponse | ErrorResponse) = await acquireToken(resourceId, userId, this.metadata.settings.clientId);
+		if (response.error) {
+			throw new Error(`Response contained error ${response}`);
+		}
+
+		response = response as TokenResponse;
+
+		context.cache.add([response], (err, result) => {
+			console.log(err, result);
```

Comment:
This is going to try and use result as a param to err which doesn't seem correct - is that the expected format?

Reply:
Nah, console.log() can either do substitution or just print out multiple things:

https://developer.mozilla.org/en-US/docs/Web/API/Console/log